x-pack/plugins/cases/public/components/custom_fields/field_type/field_type_dropdown.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { EuiSuperSelectOption } from '@elastic/eui';
+import { EuiFlexGroup, EuiSuperSelect, EuiFlexItem, EuiText } from '@elastic/eui';
+import { CustomFieldTypes } from '../types';
+import * as i18n from '../translations';
+
+interface Props {
+  disabled: boolean;
+  isLoading: boolean;
+  onChange?: (newValue: string) => void;
+  selectedType: string;
+}
+
+interface CustomFieldTypeValues {
+  id: CustomFieldTypes;
+  label?: string;
+}
+
+export const FieldTypeDropdownComponent = ({
+  disabled,
+  isLoading,
+  onChange,
+  selectedType,
+}: Props) => {
+  const customFieldTypesValues: CustomFieldTypeValues[] = [
```

Comment:
This information should be built with the use of the `builderMap`. The custom fields can define their own labels. Then you can iterate over the `builderMap` and get the information you need.